sdk/lib/core/date_time.dart

```diff
@@ -399,6 +399,24 @@ class DateTime implements Comparable<DateTime> {
   static const int _maxMicrosecondsSinceEpoch =
       _maxMillisecondsSinceEpoch * Duration.microsecondsPerMillisecond;
 
+  /// Constructs a new [DateTime] instance
+  /// with the given [secondsSinceEpoch].
+  ///
+  /// /// If [isUtc] is false then the date is in the local time zone.
+  ///
+  /// The constructed [DateTime] represents
+  /// 1970-01-01T00:00:00Z + [secondsSinceEpoch] seconds in the given
+  /// time zone (local or UTC).
+  /// ```dart
+  /// final newYearsDay =
+  ///     DateTime.fromSecondsSinceEpoch(1641031200, isUtc:true);
+  /// print(newYearsDay); // 2022-01-01 10:00:00.000Z
+  /// ```
+  external DateTime.fromSecondsSinceEpoch(
```

Comment:
I can see some logics about max value and validation for milliseconds and microseconds from epoch but nothing for seconds since epoch, is that expected ?